Content.Shared/Alert/AlertsSystem.cs

```diff
@@ -71,107 +124,153 @@ public bool TryGetAlertState(EntityUid euid, AlertKey key, out AlertState alertS
     /// </summary>
-    /// <param name="euid"></param>
+    /// <param name="entity"></param>
     /// <param name="alertType">type of the alert to set</param>
     /// <param name="severity">severity, if supported by the alert</param>
     /// <param name="cooldown">cooldown start and end, if null there will be no cooldown (and it will
     ///     be erased if there is currently a cooldown for the alert)</param>
     /// <param name="autoRemove">if true, the alert will be removed at the end of the cooldown</param>
     /// <param name="showCooldown">if true, the cooldown will be visibly shown over the alert icon</param>
-    public void ShowAlert(EntityUid euid, ProtoId<AlertPrototype> alertType, short? severity = null, (TimeSpan, TimeSpan)? cooldown = null, bool autoRemove = false, bool showCooldown = true )
+    public void ShowAlert(Entity<AlertsComponent?> entity,
+        ProtoId<AlertPrototype> alertType,
+        short? severity = null,
+        (TimeSpan, TimeSpan)? cooldown = null,
+        bool autoRemove = false,
+        bool showCooldown = true )
     {
         // This should be handled as part of networking.
         if (_timing.ApplyingState)
             return;
 
-        if (!TryComp(euid, out AlertsComponent? alertsComponent))
+        if (!_alertsQuery.Resolve(entity, ref entity.Comp, false))
             return;
 
-        if (TryGet(alertType, out var alert))
+        if (!TryGet(alertType, out var alert))
         {
-            // Check whether the alert category we want to show is already being displayed, with the same type,
-            // severity, and cooldown.
-            if (alertsComponent.Alerts.TryGetValue(alert.AlertKey, out var alertStateCallback) &&
-                alertStateCallback.Type == alertType &&
-                alertStateCallback.Severity == severity &&
-                alertStateCallback.Cooldown == cooldown &&
-                alertStateCallback.AutoRemove == autoRemove &&
-                alertStateCallback.ShowCooldown == showCooldown)
-            {
-                return;
-            }
+            Log.Error($"Unable to show alert {alertType}, please ensure this alertType has a corresponding YML alert prototype");
+            return;
+        }
 
-            // In the case we're changing the alert type but not the category, we need to remove it first.
-            alertsComponent.Alerts.Remove(alert.AlertKey);
+        // Check whether the alert category we want to show is already being displayed, with the same type,
+        // severity, and cooldown.
+        if (entity.Comp.Alerts.TryGetValue(alert.AlertKey, out var alertStateCallback) &&
+            alertStateCallback.Type == alertType &&
+            alertStateCallback.Severity == severity &&
+            alertStateCallback.Cooldown == cooldown &&
+            alertStateCallback.AutoRemove == autoRemove &&
+            alertStateCallback.ShowCooldown == showCooldown)
```

Comment:
And then this can just be if the passed alertState is equal instead of this if chain.